osquery/utils/windows/lzxpress.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/logger/logger.h>
+#include <osquery/utils/windows/lzxpress.h>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+
+namespace osquery {
+
+std::string decompressLZxpress(std::vector<char> prefetch_data,
+                               unsigned long size) {
+  static HMODULE hDLL =
+      LoadLibraryExW(L"ntdll.dll", NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
+  if (hDLL == nullptr) {
+    LOG(ERROR) << "Failed to load ntdll.dll";
+    return "Error";
+  }
+
+  typedef HRESULT(WINAPI * pRtlDecompressBufferEx)(
+      _In_ unsigned short format,
+      _Out_ unsigned char* uncompressedBuffer,
+      _In_ unsigned long uncompressedSize,
+      _In_ unsigned char* data,
+      _In_ unsigned long dataSize,
+      _Out_ unsigned long* finalSize,
+      _In_ PVOID workspace);
+
+  typedef HRESULT(WINAPI * pRtlGetCompressionWorkSpaceSize)(
+      _In_ unsigned short format,
+      _Out_ unsigned long* bufferWorkSpaceSize,
+      _Out_ unsigned long* fragmentWorkSpaceSize);
+
+  pRtlDecompressBufferEx RtlDecompressBufferEx;
+  pRtlGetCompressionWorkSpaceSize RtlGetCompressionWorkSpaceSize;
+  RtlGetCompressionWorkSpaceSize =
+      (pRtlGetCompressionWorkSpaceSize)GetProcAddress(
```

Comment:
Prefer `reinterpret_cast<>()` in this case instead of C casts

Reply:
good catch, switched to `reinterpret_cast<>()`